### PR TITLE
Adjust levels mobile track logic - DE46382

### PR DESCRIPTION
**Pull request by imcdonald-d2l**
Previously, the click events on the levels-mobile component didn't allow for any movement between `mousedown` and `mouseup` events. This causes issue where not all expected click events translate to a select/deselect of a criterion. 

To fix this issue I looked at the polymer `ontrack` gesture and found that they allow a `5px` radius for click events. Any movement outside this `5px` radius is considered a drag event. Implementing this makes working with the levels-mobile selector much more responsive and click events now select/deselect as expected even with some margin of movement. 

[Rally ticket](https://rally1.rallydev.com/#/57732444928d/custom/373260458992?detail=%2Fdefect%2F619200106301)

### CODE REVIEW

File: src/components/rubric-levels-mobile.js [hunk 1]
```diff
@@ -574,47 +575,50 @@ class RubricLevelsMobile extends composeMixins(
...
-		// when mouse moves we know the drag has been initiated
-		this._currentDragContext.elementHasBeenDragged = true;
-		const scrollOffset = this._scroll - this._currentDragContext.origin.scroll;
-		const fromEdge = e.pageX - this._currentDragContext.bounds.left + scrollOffset;
-		const clampedLevel = Math.floor(fromEdge / this._currentDragContext.stepWidth);
+		const dx = Math.abs(this._dragContext.origin.x - e.pageX);
+		const dy = Math.abs(this._dragContext.origin.y - e.pageY);
+		if ((dx > DRAG_DISTANCE || dy > DRAG_DISTANCE) && !this._dragContext.isDragEvent) {
+			this._dragContext.isDragEvent = true;
+		}
```
Comment (imcdonald-d2l) [inline src/components/rubric-levels-mobile.js:585]: This is where I check to see if the `mousemove` event has left the click `5px` radius and considered a drag event.

File: src/components/rubric-levels-mobile.js [hunk 2]
```diff
@@ -574,47 +575,52 @@ class RubricLevelsMobile extends composeMixins(
...
-		const scrollOffset = this._scroll - this._currentDragContext.origin.scroll;
-		const fromEdge = e.pageX - this._currentDragContext.bounds.left + scrollOffset;
-		const clampedLevel = Math.floor(fromEdge / this._currentDragContext.stepWidth);
+		const dx = Math.abs(this._dragContext.origin.x - e.pageX);
+		const dy = Math.abs(this._dragContext.origin.y - e.pageY);
+		if ((dx > DRAG_DISTANCE || dy > DRAG_DISTANCE) && !this._dragContext.isDragEvent) {
+			this._dragContext.isDragEvent = true;
+		} else if (dx <= DRAG_DISTANCE && dy <= DRAG_DISTANCE && !this._dragContext.isDragEvent) {
+			return;
+		}
```
Comment (imcdonald-d2l) [inline src/components/rubric-levels-mobile.js:587]: This is where I check to see if the `mousemove` event has left the click `5px` radius and considered a drag event. A lot of the other code changes are just re-naming things and removing unnecessary properties.

File: src/components/rubric-levels-mobile.js [hunk 3]
```diff
@@ -663,11 +668,6 @@ class RubricLevelsMobile extends composeMixins(
 		const dataIndex = parseInt(event.target.attributes.dataIndex.value);
 		if (dataIndex === this._selected) {
 			this._checkRadio(event.target);
-		} else if (dataIndex === 0 && this._selected !== -1) {
-			const radio = this._getRadioSibling(event.target, this._selected);
-			if (radio) {
-				this._checkRadio(radio);
-			}
```
Comment (imcdonald-d2l) [inline src/components/rubric-levels-mobile.js:670]: This check was preventing you from selecting the first criterion when another criterion was already selected. Don't know why it was added in the first place 🤷🏻‍♂️

File: src/components/rubric-levels-mobile.js [hunk 4]
```diff
@@ -18,6 +18,8 @@ import { RubricLevelsEntity } from '../entities/rubric-levels-entity.js';
 import { rubricStyles } from '../styles/rubric-styles.js';
 import { styleMap } from 'lit-html/directives/style-map.js';
 
+// radius for determining if drag has occurred
+const DRAG_DISTANCE = 5;
```
Comment (MitchelPaulin) [inline src/components/rubric-levels-mobile.js:22]: nit: I think a name like `DRAG_DISTANCE_PX` would be better to specify the units

File: src/components/rubric-levels-mobile.js (hunk 2 above)
Comment (MitchelPaulin) [inline src/components/rubric-levels-mobile.js:587]: You can simplify this
```suggestion
if (!this._dragContext.isDragEvent) {
		if (dx > DRAG_DISTANCE || dy > DRAG_DISTANCE) {
			this._dragContext.isDragEvent = true;
		} else {
			return;
		}
	}
}
```

File: src/components/rubric-levels-mobile.js [hunk 5]
```diff
@@ -859,7 +858,7 @@ class RubricLevelsMobile extends composeMixins(
 				id="${`level${index}`}"
 				class="${this._getLevelClassName(index, this._selected, this._hovered, this._focused, this._animating)}"
 				dataIndex="${index}"
-				cellHref="${criterionCellEntity?.getSelfLink()?.href || ''}"
+				cellHref="${criterionCellEntity.getSelfLink()}"
```
Comment (MitchelPaulin) [inline src/components/rubric-levels-mobile.js:861]: Is this change intentional?

File: src/components/rubric-levels-mobile.js (hunk 2 above)
Comment (devpow112) [reply src/components/rubric-levels-mobile.js:587]: Above line 589 is an extra bracket

Comment (MitchelPaulin) [reply src/components/rubric-levels-mobile.js:587]: > Above line 589 is an extra bracket

Good catch, updated it

File: src/components/rubric-levels-mobile.js (hunk 5 above)
Comment (imcdonald-d2l) [reply src/components/rubric-levels-mobile.js:861]: Yes, it was intentional. `CriterionCellEntity` has its own custom `getSelfLink()` function defined [here](https://github.com/Brightspace/d2l-rubric/blob/master/src/entities/rubric-criterion-cell-entity.js#L66) that returns a `''` by default.

**Review by MitchelPaulin — APPROVED** [review]

**Review by devpow112 — APPROVED** [review]
